File: test-platform/frontend/src/types/tool.ts
/** 平台工具目录接口返回的单个工具。 */
export interface Tool {
  id: string;
  name: string;
  description: string;
  entry_url: string;
  short_code: string;
  icon_key: string;
  category: string;
  features: string[];
  sort_order: number;
  /** 目录返回的主要来源仅用于解释，不可由前端推导授权。 */
  access_scope?: ToolAccessScope;
  access_source?: ToolAccessSource;
  project?: { id: string; code: string; name: string } | null;
  can_manage?: boolean;
  fallback_health_path?: string;
}

/** 工具列表接口响应。 */
export interface ToolListResponse {
  items: Tool[];
}

/** 工具在首页展示的健康状态。 */
export type ToolHealthState = "checking" | "healthy" | "unhealthy";

/** 平台产品层的四个稳定能力域。 */
export type CapabilityDomainId =
  | "ai-testing"
  | "automation"
  | "quality-analysis"
  | "domain-evaluation";

/** 工具在工作台中的使命、输入输出和边界定义。 */
export interface CapabilityDefinition {
  toolId: string;
  domainId: CapabilityDomainId;
  displayName: string;
  mission: string;
  scenario: string;
  inputSummary: string;
  outputSummary: string;
  boundary?: string;
  actionLabel: string;
  visualPriority: "primary" | "standard";
}

/** 后端授权目录与产品层定义合并后的页面模型。 */
export interface CapabilityViewModel extends CapabilityDefinition {
  tool: Tool;
  health: ToolHealthState;
}

/** 平台后端健康检查响应。 */
export interface PlatformToolHealthResponse {
  tool_id: string;
  status: "healthy" | "unhealthy";
  checked_at: string;
}
import type { ToolAccessScope, ToolAccessSource } from "./access";

File: test-platform/backend/app/schemas/tool.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


class ToolResponse(BaseModel):
    """前端可见的工具目录字段，不包含内部健康地址。"""

    id: str
    name: str
    description: str
    entry_url: str
    short_code: str
    icon_key: str
    category: str
    features: list[str]
    sort_order: int
    access_scope: Literal["public", "project"] = "project"
    project_id: str | None = None
    project_name: str | None = None
    access_source: str | None = None
    can_manage: bool = False

    model_config = ConfigDict(from_attributes=True)


class ToolListResponse(BaseModel):
    """工具目录列表响应。"""

    items: list[ToolResponse]


class ToolHealthResponse(BaseModel):
    """单个工具的健康探测结果。"""

    tool_id: str
    status: Literal["healthy", "unhealthy"]
    checked_at: datetime
    version: str | None = None
    revision: str | None = None
    dirty: bool | None = None
    runtime_environment: str | None = None
